test/cases/enum.ts

```diff
@@ -20,10 +21,158 @@ export var schema = {
   "additionalProperties": false
 }
 
-export var types = `interface Enum {
+export var configurations = [
+  {
+    settings: {
+      useTypescriptEnums: false
+    },
+    types: `interface Enum {
   foo: "a" | "b" | "c";
   bar: number;
   baz: {
     a: number;
   };
 }`
+  }, 
+  {
+    settings: {
+      useTypescriptEnums: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+enum Bar {
+  One = 1,
+  Two = 2,
+  Three = 3
+}
+enum Baz {
+  [object Object],
+  [object Object],
+  [object Object]
+}
+interface Enum {
+  foo: Foo;
+  bar: Bar;
+  baz: Baz;
+}`
+  },
+    {
+    settings: {
+      useTypescriptEnums: true,
+      addEnumUtils: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+class FooUtil {
```

Comment:
If there is a way, I haven't found it.  As far as I know, it's especially not possible with const enums, as the symbolic values are lost at runtime.  

If you look at the generated util class in particular, it's not so much runtime checking in the broad sense--this isn't for validation--as it is parsing and conversion unique to enums.  My requirement is that I need to be able to talk about enums in terms of their symbolic name, their integer value, and the string value of their symbolic name; right now, TS out-of-the-box only allows me to do the former at compile time, the middle at runtime, and the latter not at all.  So for me these functions will all have to exist in something approximating their current form; it just happens that their current form is easy to generate, so I opted to do that instead of maintaining them independently.